### PR TITLE
gx: update deps

### PR DESCRIPTION
**Pull request by Stebalien**
This updates a lot of libp2p deps. Do not merge yet.

~Waiting on https://github.com/libp2p/go-libp2p-kad-dht/pull/74~

### CODE REVIEW

**Comment by Stebalien** [comment]
Really: DO NOT MERGE. This definitely introduces bugs that we'll have to fix.

@whyrusleeping, @vyzo.

**Comment by vyzo** [comment]
I will rebase #4091 once it gets merged.

File: exchange/bitswap/wantmanager.go [hunk 1]
```diff
@@ -17,10 +17,9 @@ import (
 
 type WantManager struct {
 	// sync channels for Run loop
-	incoming   chan *wantSet
-	connect    chan peer.ID        // notification channel for new peers connecting
-	disconnect chan peer.ID        // notification channel for peers disconnecting
-	peerReqs   chan chan []peer.ID // channel to request connected peers on
+	incoming     chan *wantSet
+	connectEvent chan peerStatus     // notification channel for peers connecting/disconnecting
+	peerReqs     chan chan []peer.ID // channel to request connected peers on
```
Comment (Stebalien) [inline exchange/bitswap/wantmanager.go]: Technically, this bug existed before this PR. However, the whole point of the peerstream update was to fix bugs like this so I figured I might as well fix this now.

File: exchange/bitswap/wantmanager.go [hunk 2]
```diff
@@ -270,22 +273,22 @@ func (mq *msgQueue) openSender(ctx context.Context) error {
...
-	case pm.disconnect <- p:
+	case pm.connectEvent <- peerStatus{peer: p, connect: false}:
 	case <-pm.ctx.Done():
 	}
 }
 
 // TODO: use goprocess here once i trust it
 func (pm *WantManager) Run() {
-	tock := time.NewTicker(rebroadcastDelay.Get())
-	defer tock.Stop()
```
Comment (Stebalien) [inline exchange/bitswap/wantmanager.go]: I removed this ticker as it doesn't appear to be used.

**Comment by Stebalien** [comment]
Waiting on https://github.com/libp2p/go-floodsub/pull/29.

**Comment by Stebalien** [comment]
Waiting on https://github.com/whyrusleeping/go-multiplex/pull/5 and https://github.com/whyrusleeping/go-smux-multiplex/pull/1

**Comment by Stebalien** [comment]
Ok. This should be unblocked as I've simply removed the broken test cases (and will be removing go-multiplexer support in the near future as it's very buggy).

File: test/sharness/t0125-twonode.sh [hunk 3]
```diff
@@ -89,11 +89,6 @@ test_expect_success "set up tcp testbed" '
 	iptb init -n 2 -p 0 -f --bootstrap=none
 '
 
-# test multiplex muxer
-export LIBP2P_MUX_PREFS="/mplex/6.7.0"
-run_advanced_test "--enable-mplex-experiment"
-unset LIBP2P_MUX_PREFS
```
Comment (Kubuxu) [inline test/sharness/t0125-twonode.sh]: Why remove this?

Comment (Stebalien) [inline test/sharness/t0125-twonode.sh]: go-multiplexer is very broken, has been failing travis for a while, and @whyrusleeping told me to simply remove it. I've removed the test cases in this PR and will remove the code in another (so this one can move forward).

Comment (Stebalien) [inline test/sharness/t0125-twonode.sh]: (e.g., go-multiplexer is responsible for this bug https://github.com/ipfs/go-ipfs/issues/4038)

As for why I made this change in *this* PR... I was having trouble getting CI green.

Comment (Stebalien) [inline test/sharness/t0125-twonode.sh]: So..., @diasdavid says this is the *only* multiplexer that works in both go and JavaScript so we shouldn't remove it.

**Comment by Stebalien** [comment]
Nope. I've removed the multiplexer changes so this should be good to go.

**Pull request merged**